src/js/program/program-controller.js

```diff
@@ -198,21 +198,36 @@ export default class ProgramController extends Eventable {
     castVideo(castProvider, item) {
         const { model } = this;
 
+        const playlistItem = Object.assign({}, item);
+        playlistItem.starttime = model.mediaModel.get('position');
+
         const castMediaController = new MediaController(castProvider, model);
-        castMediaController.activeItem = item;
+        castMediaController.activeItem = playlistItem;
         this._setActiveMedia(castMediaController);
         // Initialize the provider last so it's setting properties on the (newly) active media model
-        castMediaController.provider.init(item);
-        model.trigger('itemReady', item);
+        castMediaController.provider.init(playlistItem);
+        model.trigger('itemReady', playlistItem);
+
     }
 
     /**
      * Stops casting. The Player is expected to restore video playback afterwards.
      * @returns {undefined}
      */
     stopCast() {
-        this.stopVideo();
-        this.mediaController = null;
+        const { mediaController, model } = this;
+        const index = model.get('item');
+        const item = model.get('playlist')[index];
+
+        item.starttime = model.mediaModel.get('position');
+        model.resetItem(item);
+
+        if (mediaController) {
+            mediaController.stop();
+            this.mediaController = null;
+        }
+
+        this.setActiveItem(index);
```

Comment:
> // The Player is expected to restore video playback afterwards
`setActiveItem` returns a promise which you could use to resume playback. See it's other uses in instream-adapter and the controller.